instrumentation/restlet/restlet-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/restlet/v1_0/RouteInstrumentation.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.restlet.v1_0;
+
+import static io.opentelemetry.javaagent.instrumentation.api.Java8BytecodeBridge.currentContext;
+import static net.bytebuddy.matcher.ElementMatchers.isMethod;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.takesArgument;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.instrumentation.api.tracer.ServerSpan;
+import io.opentelemetry.javaagent.extension.instrumentation.TypeInstrumentation;
+import io.opentelemetry.javaagent.extension.instrumentation.TypeTransformer;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.restlet.Route;
+import org.restlet.data.Request;
+
+public class RouteInstrumentation implements TypeInstrumentation {
+  @Override
+  public ElementMatcher<TypeDescription> typeMatcher() {
+    return named("org.restlet.Route");
+  }
+
+  @Override
+  public void transform(TypeTransformer transformer) {
+    transformer.applyAdviceToMethod(
+        isMethod()
+            .and(named("beforeHandle"))
+            .and(takesArgument(0, named("org.restlet.data.Request")))
+            .and(takesArgument(1, named("org.restlet.data.Response"))),
+        this.getClass().getName() + "$RouteBeforeHandleAdvice");
+  }
+
+  @SuppressWarnings("unused")
+  public static class RouteBeforeHandleAdvice {
+
+    @Advice.OnMethodEnter(suppress = Throwable.class)
+    public static void getRouteInfo(@Advice.This Route route, @Advice.Argument(0) Request request) {
+      String pattern = route.getTemplate().getPattern();
+      if (pattern == null || pattern.equals("")) {
+        return;
+      }
+
+      Context context = currentContext();
+
+      Span serverSpan = ServerSpan.fromContextOrNull(context);
+
+      if (serverSpan == null) {
+        return;
+      }
+
+      serverSpan.updateName(pattern);
```

Comment:
Now that I think of it, since Restlet can be run inside a Servlet container, we should probably use `ServerSpanNaming` (and `ServletContextPath`) to correctly update the server span name (other container/servlet instrumentations might want to do that too, and using that class ensures they all cooperate correctly).

Reply:
Sure, I'll try to change that